source/agora/common/crypto/Schnorr.d

```diff
@@ -223,6 +698,29 @@ public Signature sign (T) (
     return Signature(R, s);
 }
 
+/// Overload of 'sign' which takes a specific challenge
+private Signature signChallenge (const ref Scalar k, const ref Point R,
```

Comment:
Why is this function necessary ? `sign` should do the job. Also the extra hashing will commit to `R`, while this does not.